rnd/autogpt_server/autogpt_server/executor/manager.py

```diff
@@ -402,62 +396,89 @@ class Executor:
+
+    @classmethod
+    def on_node_executor_sigterm(cls):
+        llprint(f"[on_node_executor_sigterm {cls.pid}] ⚠️ SIGTERM received")
+        if not cls.shutdown_lock.acquire(blocking=False):
+            return  # already shutting down, no need to self-terminate
+
+        llprint(f"[on_node_executor_sigterm {cls.pid}] ⏳ Disconnecting DB...")
+        cls.loop.run_until_complete(db.disconnect())
+        llprint(f"[on_node_executor_sigterm {cls.pid}] ✅ Finished cleanup")
+        sys.exit(0)
+
     @classmethod
     @error_logged
-    def on_node_execution(cls, q: ExecutionQueue[NodeExecution], data: NodeExecution):
+    def on_node_execution(
+        cls, q: ExecutionQueue[NodeExecution], node_exec: NodeExecution
+    ):
         log_metadata = get_log_metadata(
-            graph_eid=data.graph_exec_id,
-            graph_id=data.graph_id,
-            node_eid=data.node_exec_id,
-            node_id=data.node_id,
+            graph_eid=node_exec.graph_exec_id,
+            graph_id=node_exec.graph_id,
+            node_eid=node_exec.node_exec_id,
+            node_id=node_exec.node_id,
             block_name="-",
         )
 
         execution_stats = {}
-        timing_info, _ = cls._on_node_execution(q, data, log_metadata, execution_stats)
+        timing_info, _ = cls._on_node_execution(
+            q, node_exec, log_metadata, execution_stats
+        )
         execution_stats["walltime"] = timing_info.wall_time
         execution_stats["cputime"] = timing_info.cpu_time
 
         cls.loop.run_until_complete(
-            update_node_execution_stats(data.node_exec_id, execution_stats)
+            update_node_execution_stats(node_exec.node_exec_id, execution_stats)
         )
 
     @classmethod
     @time_measured
     def _on_node_execution(
         cls,
         q: ExecutionQueue[NodeExecution],
-        d: NodeExecution,
+        node_exec: NodeExecution,
         log_metadata: dict,
         stats: dict[str, Any] | None = None,
     ):
         try:
-            cls.logger.info(
-                "Start node execution",
-                extra={
-                    "json_fields": {
-                        **log_metadata,
-                    }
-                },
+            logger.info(
+                f"Start node execution {node_exec.node_exec_id}",
+                extra={"json_fields": {**log_metadata}},
             )
-            for execution in execute_node(cls.loop, cls.agent_server_client, d, stats):
+            for execution in execute_node(
+                cls.loop, cls.agent_server_client, node_exec, stats
+            ):
                 q.add(execution)
-            cls.logger.info(
-                "Finished node execution",
-                extra={
-                    "json_fields": {
-                        **log_metadata,
-                    }
-                },
+            logger.info(
+                f"Finished node execution {node_exec.node_exec_id}",
```

Comment:
Can we generate a log prefix from log_metadata like
[graph-eid=...|graph-id=...|node-eid=...|node-id=...|block-name=] <your log here> ?

It's consistent and super helpful

Reply:
I just added the minimum of what I needed in terms of logging. I'm not going to make it more complicated than in this PR that because out of scope.